common/browser.go

```diff
@@ -513,6 +519,13 @@ func (b *Browser) IsConnected() bool {
 
 // NewContext creates a new incognito-like browser context.
 func (b *Browser) NewContext(opts goja.Value) (api.BrowserContext, error) {
+	b.contextMu.RLock()
+	browserCtx := b.context
+	b.contextMu.RUnlock()
+	if browserCtx != nil {
+		return nil, errors.New("close the existing browser context before creating a new one")
```

Comment:
Let me know how you feel about this. I could change this in this PR (was going to do it in a later PR) so that we don't allow more than one `browserContext` to be setup as we originally [wanted](https://github.com/grafana/k6-cloud/issues/1077#issuecomment-1570001658).